cmake/thirdparty/SetupHIP.cmake

```diff
@@ -15,24 +15,108 @@ find_package(HIP REQUIRED)
 message(STATUS "HIP version:      ${HIP_VERSION_STRING}")
 message(STATUS "HIP platform:     ${HIP_PLATFORM}")
 
-if("${HIP_PLATFORM}" STREQUAL "hcc" OR "${HIP_PLATFORM}" STREQUAL "amd")
-	set(HIP_RUNTIME_DEFINE "__HIP_PLATFORM_HCC__")
-elseif("${HIP_PLATFORM}" STREQUAL "nvcc")
-	set(HIP_RUNTIME_DEFINE "__HIP_PLATFORM_NVCC__")
+set(HIP_RUNTIME_INCLUDE_DIRS "${HIP_ROOT_DIR}/include")
+if(${HIP_PLATFORM} STREQUAL "hcc")
+	set(HIP_RUNTIME_DEFINES "-D__HIP_PLATFORM_HCC__")
+    find_library(HIP_RUNTIME_LIBRARIES NAMES hip_hcc libhip_hcc
+                PATHS ${HIP_ROOT_DIR}/lib
+                NO_DEFAULT_PATH
+                NO_CMAKE_ENVIRONMENT_PATH
+                NO_CMAKE_PATH
+                NO_SYSTEM_ENVIRONMENT_PATH
+                NO_CMAKE_SYSTEM_PATH)
```

Comment:
Should we use the `blt_find_libraries()` macro for these?
https://llnl-blt.readthedocs.io/en/develop/api/utility.html#blt-find-libraries

Reply:
I think the question of `blt_find_libraries` versus CMake standard `find_library` is the only remaining unresolved question here.

I have no preference either way, so I'm happy to implement it either way. From what I can tell there isn't any internal usage of `blt_find_libraries`, but there's only a few other instances of `find_library` so that's not a good argument.

@kennyweiss , @white238 , @joshessman-llnl , and @davidbeckingsale : do you want `find_library` or `bld_find_libraries`? I'll implement which ever ya'll prefer.

Reply:
@white238 @joshessman-llnl did you have thoughts on this?

Reply:
Sorry I completely forgot about this.  I am okay with this as it stands

Reply:
Sorry -- I completely missed your question. I'm also ok with it as is.